### PR TITLE
Update jupiter

### PR DESCRIPTION
**Pull request by Arrowana**
Let's replace the jupiter branch after review

https://github.com/jup-ag/stakedex-sdk/tree/previous-jupiter pushed the old branch here to avoid breaking any CI/rollback

Need to merge to third party then PR to jupiter core

### CODE REVIEW

File: common/src/pool_pair.rs [hunk 1]
```diff
@@ -299,7 +379,7 @@ where
     }
 
     fn get_swap_and_account_metas(&self, swap_params: &SwapParams) -> Result<SwapAndAccountMetas> {
-        let bridge_stake_seed = (self.clock.slot % u64::from(u32::MAX)).try_into().unwrap();
+        let bridge_stake_seed = rand::random();
```
Comment (ZhengYuTay) [inline common/src/pool_pair.rs:382]: previously it's just a clock number to stay within limit of u32

Comment (ZhengYuTay) [reply common/src/pool_pair.rs:382]: i ix just need a random number, seems weird from outside

File: common/src/pool_pair.rs [hunk 2]
```diff
@@ -331,4 +411,26 @@ where
     fn program_id(&self) -> Pubkey {
         stakedex_interface::ID
     }
+
+    fn get_accounts_len(&self) -> usize {
+        // Pick a single direction
+        1 + <dyn WithdrawStake>::accounts_len(&self.p1) + DepositStake::accounts_len(&self.p2) + 1
```
Comment (ZhengYuTay) [inline common/src/pool_pair.rs:417]: should this be one sided only? instead of combining two? like Math.max(withdraw,deposit)

**Review by ZhengYuTay — APPROVED** [review]

File: common/src/pool_pair.rs (hunk 2 above)
Comment (Arrowana) [reply common/src/pool_pair.rs:417]: It withdraw the stake then deposit it elsewhere, so all accounts will be present.

File: common/src/pool_pair.rs (hunk 1 above)
Comment (Arrowana) [reply common/src/pool_pair.rs:382]: The static seed per slot isn't great in case the user is sending a burst, instead do random and pray

**Pull request closed without merging**